### PR TITLE
Use new VTEX ID login flow

### PR DESCRIPTION
**Pull request by tiagonapoli**
#### What is the purpose of this pull request?
Use new VTEXID login flow.

#### How should this be manually tested?
Install the following version:
```
yarn global add vtex@2.105.1-beta.6
```
Check if `vtex login` is working

#### Types of changes
- [ ] Refactor (non-breaking change that only makes the code better)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

#### Chores checklist
- [x] Update `CHANGELOG.md`

### CODE REVIEW

File: src/api/clients/IOClients/external/VTEXID.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    return IOClientFactory.createClient<VTEXID>(VTEXID, customContext, customOptions)
+  }
+
+  constructor(ioContext: IOContext, opts: InstanceOptions) {
+    super('https://vtexid.vtex.com.br', ioContext, {
+      ...opts,
+      headers: {
+        // DEVELOPMENT ONLY
+        //  - the cookie routes to VTEX ID Beta env
+        //  - remove before release
```
Comment (tiagonapoli) [inline src/api/clients/IOClients/external/VTEXID.ts:18]: Note to self: Can't forget to remove this

File: src/api/clients/IOClients/external/VTEXID.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        cookie: 'vtex-commerce-env=beta',
+        'X-Forwarded-For': '127.0.0.1',
```
Comment (rnbez) [inline src/api/clients/IOClients/external/VTEXID.ts:20]: Remove this.

File: src/api/clients/IOClients/external/VTEXID.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  public startToolbeltLogin({ account, secretHash, loopbackUrl }: StartToolbeltLoginInput) {
+    const body = querystring.stringify({
+      secretHash,
+      loopbackUrl,
+    })
+
+    return this.http.post<string>(`${VTEXID.TOOLBELT_API_PATH_PREFIX}/start?an=${account}`, body)
+  }
+
+  public validateToolbeltLogin({ account, loginState, secret, ott }: ValidateToolbeltLoginInput) {
```
Comment (rnbez) [inline src/api/clients/IOClients/external/VTEXID.ts:34]: tip: you can call `loginState` as `state` if you want.

File: src/lib/auth/AuthProviders/OAuthAuthenticator/index.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      const login: string = decodedToken.sub
+      this.closeChromeTabIfMac(loginServer.loginCallbackUrl)
+      return { login, token }
+    } finally {
+      loginServer.close()
+    }
+  }
+
+  private loginUrl(account: string, loginState: string) {
+    const loginPath = `${account}.myvtex.com/_v/segment/admin-login/v1/login`
```
Comment (rnbez) [inline src/lib/auth/AuthProviders/OAuthAuthenticator/index.ts:47]: I don't think it's a good idea to hardcode the admin-login path. Some accounts don't have the app installed for some unknown reason.

Comment (tiagonapoli) [reply src/lib/auth/AuthProviders/OAuthAuthenticator/index.ts:47]: How to address this possible issue?

**Pull request merged**